src/quilla/plugins/blob_storage.py

```diff
@@ -0,0 +1,167 @@
+                cleaned up
+        '''
+        self._container_client = client = ContainerClient.from_connection_string(
+            connection_string,
+            container_name=container_name
+        )
+
+        self.max_run_days = run_days
+
+        if not client.exists():
+            client.create_container()
+
+    @property
+    def container_client(self) -> ContainerClient:
+        '''
+        An instance of the container client, casting it to ContainerClient.
+
+        This should be used exclusively from the abstract methods from BaseStorage
+
+        Returns:
+            The container client
+        '''
+
+        return cast(ContainerClient, self._container_client)
+
+    def quilla_configure(self, args: Namespace):
+        '''
+        Configures the plugin to run
+
+        Args:
+            args: A namespace generated by parsing the args from the CLI
+        '''
+        if args.connection_string is None:
+            return
+
+        self.configure(
+            args.connection_string,
+            args.container_name,
+            args.run_days,
+        )
+
+    @property
+    def is_enabled(self) -> bool:
+        return self._container_client is not None
+
+    def find_image_by_baseline(self, baseline_id: str) -> bytes:
+        blob = self.container_client.get_blob_client(f'baselines/{baseline_id}.png')
+
+        if not blob.exists():
+            return b''
+
+        downloader = blob.download_blob()
+
+        blob_data = downloader.readall()
+
+        return blob_data
+
+    def store_baseline_image(self, run_id: str, baseline_id: str, baseline: bytes) -> str:
+        blob = self.container_client.get_blob_client(f'baselines/{baseline_id}.png')
+        snapshot = self.container_client.get_blob_client(
+            f'baselines/snapshots/{run_id}/{baseline_id}.png'
+        )
+
+        blob.upload_blob(baseline)
+        snapshot.upload_blob(baseline)
+
+        return blob.url
+
+    def store_treatment_image(self, run_id: str, baseline_id: str, treatment: bytes) -> str:
+        blob = self.container_client.get_blob_client(f'runs/{run_id}/{baseline_id}_treatment.png')
+        blob.upload_blob(treatment)
+
+        return blob.url
+
+    def make_baseline_uri(self, run_id: str, baseline_id: str) -> str:
+        baseline_data = self.find_image_by_baseline(baseline_id)
+        blob = self.container_client.get_blob_client(f'runs/{run_id}/{baseline_id}.png')
+
+        blob.upload_blob(baseline_data)
+
+        return blob.url
+
+    def cleanup_reports(self):
+        blobs = self.container_client.list_blobs()
+        current_time = datetime.now()
+
+        for blob in filter(lambda x: 'runs' in x.name, blobs):
+            time_created: datetime = datetime.fromtimestamp(blob.creation_time.timestamp())
+
+            delta = current_time - time_created
+            if self.max_run_days > -1 and delta.days >= self.max_run_days:
```

Comment:
"> max days" instead of ">= max days" would make more sense?

Reply:
Oh whoops I set it to >= to test it so I could set the max run days to 0 to clean up all the reports